sessionctx/context.go

```diff
@@ -129,12 +129,25 @@ const (
 	LastExecuteDDL basicCtxType = 3
 )
 
-type connIDCtxKeyType struct{}
+type (
+	connIDCtxKeyType      struct{}
+	commitConcurrencyType struct{}
+	ttlMngLifetimeType    struct{}
+)
 
-// ConnID is the key in context.
-var ConnID = connIDCtxKeyType{}
+var (
+	// ConnID is the key in context for connection id.
+	ConnID = connIDCtxKeyType{}
+	// CommitterConcurrency is the key in context for commit concurrency.
+	CommitterConcurrency = commitConcurrencyType{}
+	// TTLMngLifetime is the key in context for ttl manager lifetime.
+	TTLMngLifetime = ttlMngLifetimeType{}
+)
 
 // SetCommitCtx sets connection id into context
 func SetCommitCtx(ctx context.Context, sessCtx Context) context.Context {
-	return context.WithValue(ctx, ConnID, sessCtx.GetSessionVars().ConnectionID)
+	ctx = context.WithValue(ctx, ConnID, sessCtx.GetSessionVars().ConnectionID)
```

Comment:
I'm always against passing information through context.Context.
We can't gurantee that the callee ignore current context or replace it with a context.Background, then the value loss.
Maybe this way has no problem when the context.Value is optional, but it's really not a good way to pass important information.

I think I should refactor the code here in the future after this PR merge.

Reply:
+1 for not use context.Context to pass data but modify Commit interface will invoke many modifications,  maybe we can fire another pr to add a typed commit context param